lib/Client.test.js

```diff
@@ -86,24 +94,22 @@ describe("Client", () => {
 
         // when
         client.start();
-        jest.advanceTimersByTime(customClientOptions.interval);
+        await advanceTimersByTime(customClientOptions.interval);
 
-        //then
+        // then
         expect(fetchAndLockSpy).toBeCalled();
-        await engineService.fetchAndLock({});
-        jest.advanceTimersByTime(customClientOptions.interval);
         expect(pollSpy).toHaveBeenCalledTimes(2);
```

Comment:
Why does this test need to be adjusted?

Reply:
**Story time:** At first, not all the tests were running successfully on my machine. Then I started to look into the tests and I noticed that we have a few `await engineService.<something>;` added, sometimes even multiple times. See here for example: https://github.com/camunda/camunda-external-task-client-js/blob/master/lib/Client.test.js#L182

I found out why these were added (and why it wasn't working on my machine), because jest can have some issues with promises: https://github.com/facebook/jest/issues/2157.

To solve this, I replaced them with another workaround from the issue above. This is the  _flushing_ of the promises [here](https://github.com/camunda/camunda-external-task-client-js/blob/bf58b598ce73add51c20caf9c2bfa40f368a8f34/lib/Client.test.js#L30). This one works in a more reliable way and I think it's also nicer than calling awaits multiple time on unrelated methods.

**In this particular test** there was no point of these two lines. I think it was only added because of the promise issue so the test was successful. When the client is started the `poll` is called immediately, then after 1x interval the `poll` will be called again. If we advanced the time again by 1x interval, then the `poll` would be called a 3rd time which should fail the `expect(pollSpy).toHaveBeenCalledTimes(2);`. Also the `await` call is unnecessary from the test perspective.